module/module.go

```diff
@@ -0,0 +1,422 @@
+package module
+
+import (
+	"context"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/fullstorydev/grpcurl"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/grpcreflect"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+	pb "go.viam.com/api/module/v1"
+	robotpb "go.viam.com/api/robot/v1"
+	"go.viam.com/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	reflectpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/protoutils"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot/client"
+	"go.viam.com/rdk/subtype"
+)
+
+// HandlerMap is the format for api->model pairs that the module will service.
+// Ex: rdk:component:motor -> [acme:marine:thruster, acme:marine:outboard].
+type HandlerMap map[resource.RPCSubtype][]resource.Model
+
+// ToProto converts the HandlerMap to a protobuf representation.
+func (h HandlerMap) ToProto() *pb.HandlerMap {
+	pMap := &pb.HandlerMap{}
+	for s, models := range h {
+		subtype := &robotpb.ResourceRPCSubtype{
+			Subtype: protoutils.ResourceNameToProto(resource.Name{
+				Subtype: s.Subtype,
+				Name:    "",
+			}),
+			ProtoService: s.SvcName,
+		}
+
+		handler := &pb.HandlerDefinition{Subtype: subtype}
+		for _, m := range models {
+			handler.Models = append(handler.Models, m.String())
+		}
+		pMap.Handlers = append(pMap.Handlers, handler)
+	}
+	return pMap
+}
+
+// NewHandlerMapFromProto converts protobuf to HandlerMap.
+func NewHandlerMapFromProto(ctx context.Context, pMap *pb.HandlerMap, conn *grpc.ClientConn) (HandlerMap, error) {
+	hMap := make(HandlerMap)
+	refClient := grpcreflect.NewClient(ctx, reflectpb.NewServerReflectionClient(conn))
```

Comment:
we should get a nice looking error for when an SDK hasn't registered reflection services

Reply:
Added a "bail early" line in the loop just below this, so it fails out with a prettier message when that's the case.